validator/validators/performance/performance_validator.go

```diff
@@ -57,3 +90,153 @@ func (s *PerformanceValidator) Cleanup() error {
+		performanceMetricResults[metricName] = metricStats
+	}
+
+	return packIntoPerformanceInformation(receiver, dataType, fmt.Sprint(agentCollectionPeriod), commitHash, commitDate, map[string]interface{}{dataRate: performanceMetricResults}), nil
+}
+
+func (s *PerformanceValidator) GetPerformanceMetrics(startTime, endTime time.Time) ([]types.MetricDataResult, error) {
+	var (
+		metricNamespace              = s.vConfig.GetMetricNamespace()
+		validationMetric             = s.vConfig.GetMetricValidation()
+		ec2InstanceId                = awsservice.GetInstanceId()
+		performanceMetricDataQueries = []types.MetricDataQuery{}
+	)
+	log.Printf("Start getting performance metrics from CloudWatch")
+	for _, metric := range validationMetric {
+		metricDimensions := []types.Dimension{
+			{
+				Name:  aws.String("InstanceId"),
+				Value: aws.String(ec2InstanceId),
+			},
+		}
+		for _, dimension := range metric.MetricDimension {
+			metricDimensions = append(metricDimensions, types.Dimension{
+				Name:  aws.String(dimension.Name),
+				Value: aws.String(dimension.Value),
+			})
+		}
+		performanceMetricDataQueries = append(performanceMetricDataQueries, s.buildStressMetricQueries(metric.MetricName, metricNamespace, metricDimensions))
+	}
+
+	metrics, err := awsservice.GetMetricData(performanceMetricDataQueries, startTime, endTime)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return metrics.MetricDataResults, nil
+}
+
+func (s *PerformanceValidator) buildStressMetricQueries(metricName, metricNamespace string, metricDimensions []types.Dimension) types.MetricDataQuery {
+	metricInformation := types.Metric{
+		Namespace:  aws.String(metricNamespace),
+		MetricName: aws.String(metricName),
+		Dimensions: metricDimensions,
+	}
+
+	metricDataQuery := types.MetricDataQuery{
+		MetricStat: &types.MetricStat{
+			Metric: &metricInformation,
+			Period: aws.Int32(10),
+			Stat:   aws.String(string(models.AVERAGE)),
+		},
+		Id: aws.String(strings.ToLower(metricName)),
+	}
+	return metricDataQuery
+}
+
+// packIntoPerformanceInformation will package all the information into the required format of MongoDb Database
+// https://github.com/aws/amazon-cloudwatch-agent-test/blob/e07fe7adb1b1d75244d8984507d3f83a7237c3d3/terraform/setup/main.tf#L8-L63
+func packIntoPerformanceInformation(receiver, dataType, collectionPeriod, commitHash string, commitDate int64, result interface{}) PerformanceInformation {
+	instanceAMI := awsservice.GetImageId()
+	instanceType := awsservice.GetInstanceType()
+
+	return PerformanceInformation{
+		"Service":          ServiceName,
+		"UseCase":          receiver,
+		"CommitDate":       commitDate,
+		"CommitHash":       commitHash,
+		"DataType":         dataType,
+		"Results":          result,
+		"CollectionPeriod": collectionPeriod,
+		"InstanceAMI":      instanceAMI,
+		"InstanceType":     instanceType,
+	}
+}
+
+func isAllValuesGreaterThanOrEqualToZero(values []float64) bool {
```

Comment:
This function exists in `metric_dimension_test.go`. Looks to be the same other than it taking in an additional `metricName string`.

I was going to suggest sharing the code.
But this is pretty darn simple I don't think it is fine to re implement here.

Reply:
yeah i was going to say the same thing, this should probably be a shared resource

Reply:
Its pretty simple and not commonly used so I have not thought of sharing the function between the two. I'm fine with create a common function between the two if you guys has a strong urge with it.